src/runtime/pythonengine.cs

```diff
@@ -158,6 +171,9 @@ public static void Initialize(IEnumerable<string> args, bool setSysArgv = true)
         {
             if (!initialized)
             {
+                // Make sure we shut down properly on app domain reload
+                System.AppDomain.CurrentDomain.DomainUnload += new EventHandler(OnDomainUnload);
```

Comment:
Shouldn't this get undone in Shutdown()?

Otherwise it seems like we'll get two Shutdowns if there's an explicit call to Shutdown: one from the explicit call, then one from the domain unload.

Reply:
good point, I will add that

Reply:
I added the requested change. Note that I can still crash upon multiple app domain reloads, so we are not out of the woods. I will push the new dll to the other repo so at least it is a bit more stable